Yosemite/Yosemite/Stores/AppSettingsStore.swift

```diff
@@ -108,11 +113,67 @@ public class AppSettingsStore: Store {
             setProductsFeatureSwitch(isEnabled: isEnabled, onCompletion: onCompletion)
         case .resetStatsVersionStates:
             resetStatsVersionStates()
+        case .setInstallationDateIfNecessary(let date, let onCompletion):
+            setInstallationDateIfNecessary(date: date, onCompletion: onCompletion)
+        case .setLastFeedbackDate(let date, let onCompletion):
+            setLastFeedbackDate(date: date, onCompletion: onCompletion)
         }
     }
 }
 
+// MARK: - General App Settings
+
+private extension AppSettingsStore {
+    /// Save the `date` in `GeneralAppSettings` but only if the `date` is older than the existing
+    /// `GeneralAppSettings.installationDate`.
+    ///
+    func setInstallationDateIfNecessary(date: Date, onCompletion: ((Result<Void, Error>) -> Void)) {
+        do {
+            let settings = try loadOrCreateGeneralAppSettings()
+
+            let installationDate = settings.installationDate ?? .distantFuture
+            guard date < installationDate else {
```

Comment:
Maybe a doing something like 
```swift
guard let date = date, date < installationDate else {
``` 
would be easier to understand, it took me a while to get what/how `.distantFuture` played in here!

Reply:
@Ecarrion I just realized that I couldn't do `guard let date = date` because it's not the `date` that is `Optional` but the `settings.installationDate`. I changed it to this instead:

```swift
            if let installationDate = settings.installationDate,
                date > installationDate {
                return onCompletion(.success(()))
            }
```

Changed in 8a4603f6bedaa5fbee68a9faa26367708bdd612f. Is that better?